docs/maintaining.rst

```diff
@@ -0,0 +1,17 @@
+===========
+Maintaining
+===========
+
+A successful production model will need to be improved from time to time.
+
+Reviewing a Ruleset Change
+==========================
+
+The points to consider when reviewing a model change:
+
+* Make sure the metrics are better. If the change involves adding samples, do a :doc:`fathom test<commands/test>` run with the old coefficients (and the new samples) as a baseline. This should result in worse metrics than the production ruleset, since you made it harder by introducing failing samples. Then compare those metrics to a new :doc:`fathom train<commands/train>` run with the new samples and any ruleset code changes. If the second metrics are better, you should adopt the new model. See :ref:`Evaluating Metrics <evaluating-metrics>` for how to compare metrics.
+
+  Ideally you can collect several samples representative of the problem you're trying to solve and distribute them across the training/validation/test sets. If you can find only one, you'll have to settle for putting it in training so the coefficients can be informed by it.
+* Make sure the "before" and "after" metrics, with commandline flags, are in the commit message to justify the change.
+* Review ruleset code changes as in a normal code review, for correctness and comprehensibility.
+* Check that any added samples have been extracted and render properly in Firefox. Use :doc:`fathom serve<commands/serve>` to make sure cross-origin policies (which are picky for ``file://`` URLs) aren't preventing the loading of subresources.
```

Comment:
Nit: I think it could be useful having separate sections for ruleset changes vs. corpus additions. In the latter I would put this point plus checking of the sample naming (assuming a naming convention is common).

Reply:
I just gave it a try, but the bullets don't divide up neatly. A big chunk of the first and the entire second are applicable to both sections. So I'll add the naming-convention point and go with that. Thank you for having a look!

Reply:
Actually, I ended up with a "If you added samples to the corpus, do these as well:" section. That worked.